react/src/views/Transfers/CreateTransferShipment/CreateTransferShipmentView.tsx

```diff
@@ -0,0 +1,178 @@
+import CreateTransferShipment, {
+  IOrganisationsAgreementsDataData,
+  ITransferShipmentFormData,
+} from "./components/CreateTransferShipment";
+
+export const ALL_ORGS_AND_BASES_WITH_TRANSFER_AGREEMENTS_QUERY = gql`
+  ${BASE_ORG_FIELDS_FRAGMENT}
+  ${TRANSFER_AGREEMENT_FIELDS_FRAGMENT}
+  query AllAcceptedTransferAgreements($baseId: ID!) {
+    base(id: $baseId) {
+      ...BaseOrgFields
+    }
+
+    transferAgreements(states: Accepted) {
+      ...TransferAgreementFields
+    }
+  }
+`;
+
+export const CREATE_SHIPMENT_MUTATION = gql`
+  mutation CreateTransferShipment(
+    $sourceBaseId: Int!
+    $targetBaseId: Int!
+    $transferAgreementId: Int!
+  ) {
+    createShipment(
+      creationInput: {
+        sourceBaseId: $sourceBaseId
+        targetBaseId: $targetBaseId
+        transferAgreementId: $transferAgreementId
+      }
+    ) {
+      id
+    }
+  }
+`;
+
+function CreateTransferShipmentView() {
+  // Basics
+  const navigate = useNavigate();
+  const { triggerError } = useErrorHandling();
+  const { createToast } = useNotification();
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+
+  // variables in URL
+  const baseId = useParams<{ baseId: string }>().baseId!;
+
+  // Query Data for the Form
+  const allFormOptions = useQuery<AllAcceptedTransferAgreementsQuery>(
+    ALL_ORGS_AND_BASES_WITH_TRANSFER_AGREEMENTS_QUERY,
+    {
+      variables: {
+        baseId,
+      },
+    },
+  );
+
+  // Mutation after form submission
+  const [createTransferShipmentMutation, createTransferShipmentMutationState] = useMutation<
+    CreateTransferShipmentMutation,
+    CreateTransferShipmentMutationVariables
+  >(CREATE_SHIPMENT_MUTATION);
+
+  // Prep data for Form
+  const allTransferAgreements = allFormOptions.data?.transferAgreements;
+
+  // eslint-disable-next-line max-len
+  const currentOrganisationLabel = `${allFormOptions?.data?.base?.organisation?.name} - ${allFormOptions?.data?.base?.name}`;
+
+  const partnerOrgsAgreementData = allTransferAgreements
+    ?.filter(
+      (agreement) =>
+        agreement.sourceOrganisation.id === globalPreferences.selectedOrganisationId?.toString() &&
+        (agreement.type === TransferAgreementType.SendingTo ||
+          agreement.type === TransferAgreementType.Bidirectional),
+    )
+    .map(
+      (agreement) =>
+        ({
+          agreementId: agreement.id,
+          specialNote: agreement.comment,
+          orgId: agreement.targetOrganisation.id,
+          orgName: agreement.targetOrganisation.name,
+          orgBases: agreement.targetBases,
+        } as IOrganisationsAgreementsDataData),
+    );
+
+  // Handle Submission
+  const onSubmitCreateShipmentForm = (createTransferShipmentData: ITransferShipmentFormData) => {
+    // Choose the agreement id based on the partner organization's base id
+    const selectedAgreement = partnerOrgsAgreementData?.find((org) =>
```

Comment:
I'm confused by the variable `org`, doesn't `partnerOrgsAgreementData` contain agreements?

Reply:
@pylipp In the shipment UI, we just have the partner org's base information, and we need to match the selected partner org's base id to the corresponding agreement id.

Reply:
@pylipp I added optional data props to the SelectField so we can add agreement data as data and remove extra logic to retrieve agreements